### PR TITLE
feat: filter data table 

### LINKED ISSUE #756: Create filterable table component
For Getting Started page https://github.com/carbon-design-system/carbon-platform/issues/754

https://www.figma.com/file/KZ6SWR2abIniIJY4RewDeA/Getting-Started-Design---Redline?node-id=10%3A13436
Screen Shot 2022-05-20 at 8 09 53 AM

- [ ] Create MDX component
- [ ] Add to storybook

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #756 

## TESTING:  
-  http://localhost:3000/designing/get-started
-  Make sure data table functionality / designs match specs

TODOs:
- [x] Sort by maintainer on first load
- [x] Add caption / disclaimer when Figma is selected
- [x] Hide maintainers other than first one

### CODE REVIEW

File: services/web-app/components/filterable-design-kit-table/filterable-design-kit-table.module.scss [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Copyright IBM Corp. 2021, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+@use '@carbon/react/scss/breakpoint' as breakpoint;
+@use '@carbon/react/scss/spacing' as spacing;
+@use '@carbon/react/scss/theme' as theme;
+@use '@carbon/react/scss/utilities/convert' as convert;
```
Comment (alisonjoseph) [inline services/web-app/components/filterable-design-kit-table/filterable-design-kit-table.module.scss:11]:
```suggestion
```

We should be able to remove these

File: services/web-app/components/filterable-design-kit-table/filterable-design-kit-table.module.scss [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright IBM Corp. 2021, 2022
```
Comment (alisonjoseph) [inline services/web-app/components/filterable-design-kit-table/filterable-design-kit-table.module.scss:2]:
```suggestion
 * Copyright IBM Corp. 2022, 2022
```

File: services/web-app/components/filterable-design-kit-table/filterable-design-kit-table.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright IBM Corp. 2021, 2022
```
Comment (alisonjoseph) [inline services/web-app/components/filterable-design-kit-table/filterable-design-kit-table.js:2]:
```suggestion
 * Copyright IBM Corp. 2022, 2022
```

File: services/web-app/components/filterable-design-kit-table/index.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright IBM Corp. 2021, 2022
```
Comment (alisonjoseph) [inline services/web-app/components/filterable-design-kit-table/index.js:2]:
```suggestion
 * Copyright IBM Corp. 2022, 2022
```

File: services/web-app/data/nav-data.js [hunk 5]
```diff
@@ -29,6 +29,10 @@ export const standardsNavData = [
 ]
 
 export const assetsNavData = [
+  {
+    path: '/filter-data-table-test',
+    title: 'Filter Datatable Test'
+  },
```
Comment (alisonjoseph) [inline services/web-app/data/nav-data.js:35]:
```suggestion
```
I don't think we need this?

File: services/web-app/components/filterable-design-kit-table/filterable-design-kit-table.js [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+  })
+
+  // alphabetically sort maintainer
+  const orderedDesignKits = designKits.sort((a, b) =>
+    a.maintainer?.toLowerCase() < b.maintainer?.toLowerCase()
+      ? -1
+      : b.maintainer?.toLowerCase() > a.maintainer?.toLowerCase()
+        ? 1
+        : 0
+  )
```
Comment (alisonjoseph) [inline services/web-app/components/filterable-design-kit-table/filterable-design-kit-table.js:66]: linting is yelling about this

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
This is looking good! Couple small things

File: services/web-app/components/filterable-design-kit-table/filterable-design-kit-table.js [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+  const handleFilterChange = ({ selectedItem }) => {
+    setCurrentItem(selectedItem)
+  }
+
+  useEffect(() => {
+    setFilteredRows(filterByDesignTool(orderedDesignKits))
+  }, [currentItem, orderedDesignKits, filterByDesignTool])
+
+  // only render the firt maintainer within the respective group
```
Comment (alisonjoseph) [inline services/web-app/components/filterable-design-kit-table/filterable-design-kit-table.js:96]:
```suggestion
  // only render the first maintainer within the respective group
```

**Review by alisonjoseph — APPROVED** [review]
LGTM

**Review by francinelucca — CHANGES_REQUESTED** [review]
visuals LGTM, CR suggestion and there seems to be a re-render error, see browser console: 

image

Could be related to useEffect(), not sure what the specific issue is here

**Review by francinelucca — APPROVED** [review]
👯‍♀️

**Pull request merged**